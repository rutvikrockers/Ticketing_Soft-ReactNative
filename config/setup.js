// create our store for testing redux components
import configureMockStore from 'redux-mock-store';
const thunk = require('redux-thunk');

// Setup adapter to work with enzyme 3.2.0
const Enzyme = require('enzyme');
const Adapter = require('enzyme-adapter-react-16');

export const mockStore = configureMockStore([thunk.default]);

Enzyme.configure({ adapter: new Adapter() });