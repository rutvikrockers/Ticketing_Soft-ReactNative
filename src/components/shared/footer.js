import React, { Component } from 'react';
import {
  View,
} from 'react-native';
import {connect} from 'react-redux';
import Icon from 'react-native-vector-icons/dist/MaterialIcons';
import {Link} from 'react-router-native';

import styles from '../../styles/main.js';

class Footer extends Component {
  constructor(props) {
    super(props);
  }
  render() {
    return (
      <View style={styles.footer}>
        {/* <Link to="/menu"><Icon size={30} name="explore" color="white" style={styles.footerBtn} /></Link>
        <Link to="/"><Icon size={30} name="menu" color="white" style={styles.footerBtn} /></Link>
        <Link to="/menu"><Icon size={30} name="mail-outline" color="white" style={styles.footerBtn} /></Link>
        <Link to="/profile"><Icon size={30} name="account-circle" color="white" style={styles.footerBtn} /></Link> */}
      </View>
    )
  }
}

const mapStateToProps = (state) => {
  return {
  }
}

export default connect(mapStateToProps)(Footer);

