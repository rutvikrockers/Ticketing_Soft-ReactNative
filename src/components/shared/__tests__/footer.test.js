import React from 'react';
import { shallow } from 'enzyme';
import { Provider } from 'react-redux';
import { mockStore } from '../../../../config/setup';
import toJson from 'enzyme-to-json';
import { default as Footer } from '../footer';

const render = (props) => shallow(
    <Provider store={mockStore({})}>
        <Footer {...props} />
    </Provider>
);

describe('Footer', () => {
    let component;
    beforeEach(() => {
        component = render({});
    });

    it('matches its snapshot', () => {
        expect(toJson(component, { mode: 'deep' })).toMatchSnapshot();
    });

    it('renders', () => {
        expect(component.find(Footer).length).toBe(1);
    });
});