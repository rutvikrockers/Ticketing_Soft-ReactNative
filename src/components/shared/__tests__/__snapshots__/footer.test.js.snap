// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer matches its snapshot 1`] = `<Connect(Footer) />`;
